### PR TITLE
ping_check added

### PR DESCRIPTION
**Pull request by ndushay**
this can be useful for socket connections;  it might also be subclassed by other checks, such as an action_mailer_check I have in mind.

### CODE REVIEW

**Comment by anfleene** [comment]
Thanks for the Pull request.

One question I have is that I assume you're adding this because the thing you're checking doesn't speak http correct? We already have a [Http Check](https://github.com/sportngin/okcomputer/blob/master/lib/ok_computer/built_in_checks/http_check.rb)

**Comment by ndushay** [comment]
yes, the idea is that you may want to check a connection that isn't http.

File: lib/ok_computer/built_in_checks/ping_check.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    def initialize(host, port, request_timeout = 5)
+      raise ArgumentError if host.blank? || port.blank?
+      self.host = host
+      self.port = port
+      self.request_timeout = request_timeout.to_i
+    end
+
+    # Public: Return the status of the Ping check
+    def check
+      if tcp_socket_request
```
Comment (pbyrne) [inline lib/ok_computer/built_in_checks/ping_check.rb]: You probably don't need the conditional here, right? Any failure seems to raise an exception, so I _think_ it should be sufficient to call it and then call `mark_message` afterward.

Comment (ndushay) [inline lib/ok_computer/built_in_checks/ping_check.rb]: Good point.  Just pushed a correction.

**Comment by pbyrne** [comment]
Looks good to me!

**Pull request merged**